docs/madr/decisions/072-computed-api-endpoints.md

```diff
@@ -0,0 +1,49 @@
+# Computed API Endpoints
+
+* Status: accepted
+
+## Context and Problem Statement
+
+There are two types of API endpoints in Kuma:
+
+- Resource endpoints - These are endpoints that leads to direct reads to the database. They map to Kuma resources. For example: `GET /mesh/{mesh}/dataplanes/{dataplane}`.
+- Computed endpoints - These are endpoints that provide computed data usually using existing resources. For example: `GET /mesh/{mesh}/dataplanes/{dataplane}/_overview` 
+  or `GET /mesh/{mesh}/dataplanes/{dataplane}/_metrics`.
+
+Computed endpoints represent runtime and composed state, this state is not stored in the database and there's no way to update it from the API.
+
+This MADR provides a convention for computed endpoints. The goal is to provide a consistent way to define and document computed endpoints.
+
+## Design
+
+Computed endpoints should be always prefixed with `_`, for example: `GET /mesh/{mesh}/dataplanes/{dataplane}/_overview`.
+
+This has the following benefits:
+
+- It is easy to identify computed endpoints.
```

Comment:
Kinda rhetorical question after confirmation more than anything, but this is an internal concern right?

Some other comments might touch on this a little more (not read the rest yet), and I know this MADR is somewhat for documentation of a historical decision, but curious to get comment on an opinion that from a consumers standpoint the `_` could be seen as exposing internals in the abstraction? i.e. how come a consumer becomes aware/needs to be aware of whether a URL refers to a computed result or not? 

I know this is a pretty well baked approach that we do in our HTTP API but I've always wanted to ask/hear thoughts on that ☝️ , figured now is probably good time to ask.

Reply:
> how come a consumer becomes aware/needs to be aware of whether a URL refers to a computed result or not?

If you're unfamiliar with the API you might be tempted to try and manually modify the computed resource (which is not possible).

Reply:
(btw don't anybody wait for this convo to be resolved before merging!)